Service Locator mark II. Stores

ctarda opened the pull request:
Fixes #1199 

⚠️  This PR depends on #1198 . The base branch would need to be updated after #1198 is merged ⚠️ 

as mentioned in p99K0U-1BK-p2, we are going to abstract the global dependencies behind a Service Locator.

The scope for this PR is:
- [ ] Provide access to StoresManager through the service locator. StoresManager is not a singleton anymore
- [ ] Integrate the change in call sites to StoresManager

**Note:** 
I don't like the naming I have used in this PR. I think the protocol should be named `StoresManager` instead of `Stores` and the default implementation should be `DefaultStoresManager` instead of just `StoresManager`. But, I didn't want to submit any changes to the `StoresManager` class that made review more difficult. If/when this gets approved we can rename both the interface and the class before merging.

## Changes
- Added a `Stores` protocol (see 🆙 for more context about the name)
- Added a property to the ServiceLocator to provide a reference to the Stores protocol, initialised with StoresManager
- Update all the call sites

## Testing
- Build and run the tests. In theory, because the `StoresManager.shared` method was removed from the app, if the code builds, we should be good to go. But...
- Play around with the app. it should be possible to log in and out of it.

mindgraffiti commented on WooCommerce/Classes/ServiceLocator/Stores.swift:
```diff
@@ -0,0 +1,50 @@
+import Foundation
+import Yosemite
+
+/// Abstreacts the Stores coordination
```
```suggestion
/// Abstracts the Stores coordination
```

ctarda commented:
Thanks folks. I incorporated all your feedback, and renamed StoresManager to DefaultStoresManager and Stores to StoresManager, so our client code is still consuming a StoresManage as it used to. I'll merge as soon as the build passes
